src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -123,7 +123,19 @@ public static void RegisterInstance(VisualStudioInstance instance)
                 ApplyDotNetSdkEnvironmentVariables(instance.MSBuildPath);
             }
 
-            RegisterMSBuildPath(instance.MSBuildPath);
+            if (instance.DiscoveryType == DiscoveryType.VisualStudioSetup || instance.DiscoveryType == DiscoveryType.DeveloperConsole)
+            {
+                RegisterMSBuildPath(new string[]
+                {
+                    instance.MSBuildPath,
+                    // Find and load NuGet assemblies if msbuildPath is in a VS installation
+                    Path.GetFullPath(Path.Combine(instance.MSBuildPath, "..", "..", "..", "Common7", "IDE", "CommonExtensions", "Microsoft", "NuGet"))
```

Comment:
Should this be conditional on it existing?

Reply:
Specifically we probably want this for corext type discovery too.

Reply:
That was the problem with the previous spot, but it should always exist if we're in VS or a developer console, right?

Reply:
Corext type discovery? I only see three options (pre-mono): developer console, VS, and dotnet sdk.

Reply:
#84. I think it'd be easiest to do the exists check but we could extend the if there as well.

Reply:
Good point—forgot about that PR. Fixed!